### ISSUE TITLE
Can't bind actions on alt+mouse4/5 when using logi options+ with mx master 3s

### ISSUE THREAD

**Issue by shodanx2** [RamonUnch/AltSnap#494]
But without that software the mouse is very annoying to use

There is also a thumb button on the side, don't know what's it called but no way to bind actions to that in alt-snap either

![2023-11-19 08_30_32-MX Master 3S Wireless Mouse _ Logitech Canada — Mozilla Firefox](https://github.com/RamonUnch/AltSnap/assets/10621885/e25dfe45-db43-49af-81ac-24beea71aacf)

In logi options+ you can set those buttons to "do nothing" but then they really do nothing, they don't do mouse button 4/5

But there also isn't an option to make them regular button 4/5

Only thing I could do it bind keypress to them, so I gave them ctrl+pageup ctrl+pagedown, but that also won't bind.

**Comment by RamonUnch** [comment]
> But there also isn't an option to make them regular button 4/5

Maybe it is named next/previous page instead of button 4/5 inside the software (like in the case of my Zealot12 mouse.
anyway if you are able to bind a mouse button to a keyboard key that you have no use for, you can then add the said vKey code to the XXbutton list, and it will map to MB6,7 etc...
Look at the wiki for more details.
https://github.com/RamonUnch/AltSnap/wiki#use-more-than-five-mouse-buttons

**Comment by shodanx2** [comment]
Hi,

I tried making this work some more.

I was mistaken, I thought that the Alt-snap keyboard shortcut all needed "alt" to enable alt-snap action override, but it needs to be added to the  keyboard shortcut

First here is how my mouse is configured

![2023-11-19 17_57_22-Logi Options+](https://github.com/RamonUnch/AltSnap/assets/10621885/20fdadaf-0fef-4978-89cf-4459714a7002)

For test, I tried to setup AltSnap as follows

![2023-11-19 18_03_04-_new 15 - Notepad++](https://github.com/RamonUnch/AltSnap/assets/10621885/c92af223-e965-4484-833e-7619cee57b52)

I click save, I click apply, I click ok

Then I re-open AltSnap configuration window

![2023-11-19 18_04_13-_new 15 - Notepad++](https://github.com/RamonUnch/AltSnap/assets/10621885/79d5d5da-30b9-4ba1-b4dd-b675d03e2fae)

### But the changes don't seem to stick

I tried "clear key", save, ok , then I close AltSnap


![2023-11-19 18_07_32-](https://github.com/RamonUnch/AltSnap/assets/10621885/f652ea62-b3bb-4d6a-a4a1-be5b4f4ac10b)



![2023-11-19 18_07_14-_new 15 - Notepad++](https://github.com/RamonUnch/AltSnap/assets/10621885/39c8ee26-ff5d-4495-80f0-8d760c68e683)

But the key is still bound, I think this is what confused me earlier

![2023-11-19 18_08_30-_new 15 - Notepad++](https://github.com/RamonUnch/AltSnap/assets/10621885/b80dd292-8315-4e84-8aee-9382ee4801a3)

In this current configuration, (what I believe is) "mouse 4" does do the action "roll/unroll window" with no modifier key

![2023-11-19 18_10_19-Logi Options+](https://github.com/RamonUnch/AltSnap/assets/10621885/80213f40-c6ac-432f-80c4-29c3dbbf8a02)



My end goal for now, make Alt-mouse-4 do window maximize of the pointed window, (plain mouse 4 should still do ctrl+pageup).


So basically, I want the regular Alt-right-click behaviour on my Mouse 4 button, this is because I want alt-right-click to do the same action as titlebar-drag-right-click which for me is "enable fancyzones"

**Comment by RamonUnch** [comment]
> My end goal for now, make Alt-mouse-4 do window maximize of the pointed window, (plain mouse 4 should still do ctrl+pageup).

If Mouse 4 is Ctrl+PageUp, then Alt+Mouse 4 should be Ctrl+Alt+PageUp that you can bind to the said action?

> So basically, I want the regular Alt-right-click behaviour on my Mouse 4 button, this is because I want alt-right-click to do the same action as titlebar-drag-right-click which for me is "enable fancyzones"

I am not sure I understand what you want to do. AltSnap cannot trigger FancyZone because FancyZone hooks the window event and needs a real click inside the titlebar in order to snap it the window. AltSnap does not generate this kind of click.

AltSnap cannot see the difference between keys that come from your mouse and your keyboard. So if you set MB4=Ctrl+PageUp and setup AltSnap to use Ctrl+PageUp shortcut, then the shortcut will always be used by AltSnap, overriding all application specific shortcuts.

If you want to trigger AltSnap movement with MB4 you need to map MB4 to something else than a keyboard shortcut, because you cannot start/end AltSnap movement with a shortcut. You need in this case to map MB4 to a simple key that will generate a vKcode on press and on release.

Whereas if you bind MB4 to Ctrl+PageUp, it will generate instantly keys press and releases on MB4.down, without generating anything on MB4.up.

**Comment by shodanx2** [comment]
Hello, 

I have badly asked this question, I will reformulate in a new issue later.

I now use Xmousebuttoncontrol in conjunction with AltSnap as Xmousebuttoncontrol as per-application button remapping.

They are great complements in my opinion.

However Xmousebuttoncontrol is proprietary so I will try to switch to Ymousebuttoncontrol eventually
